### DISCUSSION TITLE
Idiomatic way to set response status code in server route

### DISCUSSION THREAD

**Discussion by ohmree** [nuxt/nuxt#17449] [Questions]
So far I found out I can set `res.statusCode` and then return `res.end()` or use (abuse?) h3's `createError` with a non-error status code but neither way feels too idiomatic.

What's the recommended way to set the response's status code from an api route?

**Comment by Flanker72** [comment]
You can simply use `event` wrapper passed in the `defineEventHandler` like this:

```
export default defineEventHandler((event) => {
  event.res.statusCode = 401;
  return { message: "Unauthorized access" };
});
```

**Comment by wobsoriano** [comment]
https://nuxt.com/docs/api/utils/set-response-status
